main.go

```diff
@@ -212,17 +216,28 @@ func printResults(results []*RunResults, totals *TotalResults, format string) {
 	}
 }
 
-func generateTLSConfig(certFile string, keyFile string) *tls.Config {
+func generateTLSConfig(certFile string, keyFile string, caFile string, insecure bool) *tls.Config {
 	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
 		log.Fatalf("Error reading certificate files: %v", err)
 	}
 
+	var caCertPool *x509.CertPool = nil
+	if caFile != "" {
+		caCert, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			log.Fatalf("Error reading CA certificate file: %v", err)
+		}
+		caCertPool = x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```

Comment:
Add a check that the certificate was loaded successfully ([AppendCertsFromPEM()]( https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM) would return false if not)?